server/routes/user.js

```diff
@@ -0,0 +1,158 @@
+import { Router } from 'express'
+import ensureAuthenticated from '../utils/passport/ensure-authenticated'
+
+const router = Router()
+
+router
+  .route(`/api/v1/users/:uid/site-subjects`)
+  .get(ensureAuthenticated, async (req, res) => {
+    try {
+      const { prisma } = req.app.locals
+      const { siteSubjects } = await prisma.users.findUnique({
+        where: { uid: req.params.uid },
+        select: { siteSubjects: true },
+      })
+
+      return res.status(200).json({ data: siteSubjects })
+    } catch (error) {
+      console.error(error.stack)
+
+      return res.status(500).send({ message: error.message })
+    }
+  })
+
+router
+  .route('/api/v1/user/:uid/site-subjects/star')
+  .post(ensureAuthenticated, async (req, res) => {
+    try {
+      const { prisma } = req.app.locals
+      const user = await prisma.users.findUnique({
+        where: { uid: req.params.uid },
+        select: {
+          siteSubjects: true,
+        },
+      })
+      const { siteSubjects } = user
+      const siteSubjectSiteIndex = siteSubjects.findIndex(
+        (siteSubject) => siteSubject.site === req.body.site
+      )
+      if (siteSubjectSiteIndex > -1) {
+        siteSubjects[siteSubjectSiteIndex].starredSubjects.push(
+          req.body.subject
+        )
+      } else {
+        siteSubjects.push({
+          site: req.body.site,
+          starredSubjects: [req.body.subject],
+        })
+      }
+      await prisma.users.update({
+        where: { uid: req.params.uid },
+        data: { siteSubjects },
+      })
+
+      return res.status(200).json({ message: 'Added Subject' })
+    } catch (error) {
+      console.error(error.stack)
+
+      return res.status(500).send({ message: error.message })
+    }
+  })
+  .delete(ensureAuthenticated, async (req, res) => {
+    try {
+      const { prisma } = req.app.locals
+      const user = await prisma.users.findUnique({
+        where: { uid: req.params.uid },
+        select: {
+          siteSubjects: true,
+        },
+      })
+      const { siteSubjects } = user
+      const siteSubjectSiteIndex = siteSubjects.findIndex(
+        (siteSubject) => siteSubject.site === req.body.site
+      )
+      siteSubjects[siteSubjectSiteIndex].starredSubjects = siteSubjects[
+        siteSubjectSiteIndex
+      ].starredSubjects.filter((subject) => subject !== req.body.subject)
+
+      const updatedUser = await prisma.users.update({
+        where: { uid: req.params.uid },
+        data: { siteSubjects },
+      })
+      console.dir(updatedUser, { depth: null })
```

Comment:
did you mean to leave this in?

Reply:
https://github.com/AMP-SCZ/dpdash/pull/241/commits/72c05efb5af2f44f7bf7b1f54d195f4143124b07